### PR TITLE
[VEL-2833] Reintroduce portal-based select updates

### PR DESCRIPTION
**Pull request by phndiaye**
This reverts commit fe847cd33f2f4c367cbeecfc6cbb20cfbc995f31, reversing changes made to 7ccf9e6ef54241ea4c69cc87e1aaea10ebeff3b6.

### What does this PR do?

Reintroduce `OSS::Select` & `OSS::PowerSelect` updates w/ portaled-dropdowns and:
- automatic positioning
- exclusive dropdowns (prevent having two dropdowns displayed at once — this only concerns those extending the `BaseDropdown` class) 
- CSS-targetable dropdowns via the `addressableAs` arg (to style them easily despite them being renderered in a portal) 

Compared to the initial implementation, I finally ended up letting the `div` elements instead of using `details/summary`. However, I decided to replicate the `open` attribute that came in handy and is added on the component when the dropdown is being displayed.
To keep them in sync and update the `isOpen` tracked prop accordingly, I use a `MutationObserver` which allows us to track whether or not the `open` attribute is present or not.

### Good PR checklist

- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [x] Added/updated tests
- [x] Added/updated documentation
- [ ] Migrated touched components to Glimmer Components
- [x] Properly labeled

### CODE REVIEW

File: addon/components/o-s-s/power-select.ts [hunk 1]
```diff
@@ -12,20 +16,33 @@ interface OSSPowerSelectArgs {
...
 const DEFAULT_PLACEHOLDER = 'Select an item';
 
-export default class OSSPowerSelect extends Component<OSSPowerSelectArgs> {
-  @tracked displaySelect: boolean = false;
+export default class OSSPowerSelect extends BaseDropdown<OSSPowerSelectArgs> {
+  cleanupDrodpownAutoplacement?: () => void;
+
+  portalId: string = guidFor(this);
+
+  declare portalTarget: HTMLElement;
```
Comment (JulienVannier66) [inline addon/components/o-s-s/power-select.ts:32]: Same in OSS::Select maybe move it to the `BaseDropdown`

File: addon/components/o-s-s/power-select.ts [hunk 2]
```diff
@@ -12,20 +16,33 @@ interface OSSPowerSelectArgs {
...
     return this.args.placeholder ?? DEFAULT_PLACEHOLDER;
   }
 
+  get dropdownAddressableClass(): string {
+    return this.args.addressableAs ? `${this.args.addressableAs}__dropdown` : '';
+  }
+
+  noop(event: Event): void {
+    event.stopPropagation();
+  }
```
Comment (JulienVannier66) [inline addon/components/o-s-s/power-select.ts:44]: Maybe move also this method in the `BaseDropdown` too

File: addon/components/o-s-s/power-select.ts [hunk 3]
```diff
@@ -38,19 +55,48 @@ export default class OSSPowerSelect extends Component<OSSPowerSelectArgs> {
...
+    super.onClickOutside(_, event);
+    this.cleanupDrodpownAutoplacement?.();
     this.args.onSearch?.('');
+    document.querySelector(`#${this.portalId}`)?.remove();
+  }
+
+  @action
+  registerContainer(element: HTMLElement): void {
+    super.registerContainer(element);
+    this.portalTarget = isTesting() ? this.container : document.body;
```
Comment (JulienVannier66) [inline addon/components/o-s-s/power-select.ts:100]: Same here too

**Review by JulienVannier66 — APPROVED** [review]

File: addon/components/o-s-s/select.stories.js [hunk 4]
```diff
@@ -103,6 +103,15 @@ export default {
           summary: 'onSearch(keyword: string): void'
         }
       }
+    },
+    addressableAs: {
+      description:
+        'A string to use as base to target the portaled-dropdown component in CSS. When provided, you will be able to use `${addressableAs}__dropdown` in CSS.',
```
Comment (Miexil) [inline addon/components/o-s-s/select.stories.js:109]: same big nit ^^
a string identifier ?

File: addon/components/o-s-s/private/base-modal.ts [hunk 5]
```diff
@@ -40,6 +41,10 @@ export default class BaseModal extends Component<BaseModalArgs> {
       document.body.append(this._parent);
     }
     this._elem = elem;
+
+    window.addEventListener('mousedown', (event) => {
```
Comment (Miexil) [inline addon/components/o-s-s/private/base-modal.ts:45]: I see addEventListener but I don't see the removeEventListener to cleanup :)

File: addon/components/o-s-s/power-select.hbs [hunk 6]
```diff
@@ -12,34 +17,55 @@
...
+        <OSS::InfiniteSelect
+          @items={{@items}}
+          @onSearch={{@onSearch}}
+          @inline={{false}}
+          @onSelect={{this.onSelect}}
+          @searchPlaceholder={{@searchPlaceholder}}
+          @loading={{@loading}}
+          @loadingMore={{@loadingMore}}
+          @onBottomReached={{@onBottomReached}}
+          class={{concat "margin-top-px-0 upf-power-select__dropdown " this.dropdownAddressableClass}}
```
Comment (Miexil) [inline addon/components/o-s-s/power-select.hbs:36]: nit:
```
class="magin-top-px-0 upf-power-select__dropdown {{this.dropdownAddressableClass}}"
```

File: addon/components/o-s-s/power-select.hbs [hunk 7]
```diff
@@ -12,34 +17,55 @@
...
```
Comment (Miexil) [inline addon/components/o-s-s/power-select.hbs:59]: nit:
```
class="magin-top-px-0 upf-power-select__dropdown {{this.dropdownAddressableClass}}"
```

File: addon/components/o-s-s/power-select.hbs (hunk 6 above)
Comment (Miexil) [reply addon/components/o-s-s/power-select.hbs:36]: less verbose

Comment (phndiaye) [reply addon/components/o-s-s/power-select.hbs:36]: if find it less visible that's why i'm not doing it like that anymore 😅

File: addon/components/o-s-s/private/base-modal.ts (hunk 5 above)
Comment (phndiaye) [reply addon/components/o-s-s/private/base-modal.ts]: Shouldn't have pushed this in the first place. removing

**Review by Miexil — APPROVED** [review]

**Pull request merged**